apps/aecore/src/aec_headers.erl

```diff
@@ -648,3 +667,22 @@ validate_max_time({Header, _}) ->
         false ->
             {error, block_from_the_future}
     end.
+
+%% TODO: this is taken from ws_int_dispatch.erl - needs refactoring after
+%% separation of key/micro block records.
+decode_block_hash(Hash) ->
```

Comment:
Move to base58c as special safe_decode and let it return {ok, ....}. Then the local decode can take away the ok.